### ISSUE TITLE
Using Touchscreen Constantly Reloads Hyprland Config & Stops Plugins

### ISSUE THREAD

**Issue by Welsey0** [hyprwm/Hyprland#8166]
### Already reported ? *

- [X] I have searched the existing open and closed issues.

### Regression?

No

### System Info and Version

<details> <summary>System/Version info</summary>

**Based upon [ML4W's Hyprland Starter](https://github.com/mylinuxforwork/hyprland-starter)**
**(Username & hostname censored for privacy)**

```sh
Hyprland, built from branch  at commit 4520b30d498daca8079365bdb909a8dea38e8d55  (version: bump to 0.44.1).
Date: Wed Oct 9 12:54:39 2024
Tag: v0.44.1, commits: 5309
built against aquamarine 0.4.2


flags: (if any)


System Information:
System name: Linux
Node name: ******-SLGo
Release: 6.11.3-arch1-1
Version: #1 SMP PREEMPT_DYNAMIC Thu, 10 Oct 2024 20:11:06 +0000


GPU information:
00:02.0 VGA compatible controller [0300]: Intel Corporation Iris Plus Graphics G1 (Ice Lake) [8086:8a56] (rev 07) (prog-if 00 [VGA controller])


os-release: NAME="EndeavourOS"
PRETTY_NAME="EndeavourOS"
ID="endeavouros"
ID_LIKE="arch"
BUILD_ID=rolling
ANSI_COLOR="38;2;23;147;209"
HOME_URL="https://endeavouros.com"
DOCUMENTATION_URL="https://discovery.endeavouros.com"
SUPPORT_URL="https://forum.endeavouros.com"
BUG_REPORT_URL="https://forum.endeavouros.com/c/general-system/endeavouros-installation"
PRIVACY_POLICY_URL="https://endeavouros.com/privacy-policy-2"
LOGO="endeavouros"


plugins:

======Config-Start======


Config File: /home/******/.config/hypr/hyprland.conf: Read Succeeded

#  _   _                  _                 _
# | | | |_   _ _ __  _ __| | __ _ _ __   __| |
# | |_| | | | | '_ \| '__| |/ _` | '_ \ / _` |
# |  _  | |_| | |_) | |  | | (_| | | | | (_| |
# |_| |_|\__, | .__/|_|  |_|\__,_|_| |_|\__,_|
#        |___/|_|
#
# -----------------------------------------------------
# Full documentation https://wiki.hyprland.org

source = ~/.config/hypr/conf/monitor.conf
source = ~/.config/hypr/conf/autostart.conf
source = ~/.config/hypr/conf/cursor.conf
source = ~/.config/hypr/conf/environments.conf
source = ~/.config/hypr/conf/input.conf
source = ~/.config/hypr/conf/general.conf
source = ~/.config/hypr/conf/decoration.conf
source = ~/.config/hypr/conf/animations.conf
source = ~/.config/hypr/conf/layouts.conf
source = ~/.config/hypr/conf/gestures.conf
source = ~/.config/hypr/conf/misc.conf
source = ~/.config/hypr/conf/windowrules.conf
source = ~/.config/hypr/conf/binds.conf


Config File: /home/******/.config/hypr/conf/monitor.conf: Read Succeeded
# See https://wiki.hyprland.org/Configuring/Monitors/
monitor= eDP-1,1536x1024,auto,1

Config File: /home/******/.config/hypr/conf/autostart.conf: Read Succeeded
# Execute your favorite apps at launch
exec-once = waybar
exec-once = swww-daemon
exec-once = swaync
exec-once = ~/.config/custom/swayosd-server.sh
exec-once = /usr/lib/polkit-gnome/polkit-gnome-authentication-agent-1
exec-once = wal -R
exec-once = hypridle
exec = hyprpm enable Hyprspace
exec-once = mpris-proxy
exec-once = wl-paste --watch cliphist store

# Load configuration from ML4W Hyprland Settings App
exec = ~/.config/ml4w-hyprland-settings/hyprctl.sh

Config File: /home/******/.config/hypr/conf/cursor.conf: Read Succeeded
env = XCURSOR_SIZE,24
env = XCURSOR_THEME,Bibata-Modern-720f0-v1.0.2-x11

Config File: /home/******/.config/hypr/conf/environments.conf: Read Succeeded
# XDG Desktop Portal
env = XDG_CURRENT_DESKTOP,Hyprland
env = XDG_SESSION_TYPE,wayland
env = XDG_SESSION_DESKTOP,Hyprland

# QT
env = QT_QPA_PLATFORM,wayland;xcb
env = QT_QPA_PLATFORMTHEME,qt6ct
env = QT_WAYLAND_DISABLE_WINDOWDECORATION,1
env = QT_AUTO_SCREEN_SCALE_FACTOR,1

# GTK
env = GDK_SCALE,1

# Mozilla
env = MOZ_ENABLE_WAYLAND,1

# Set the cursor size for xcursor
env = XCURSOR_SIZE,24

# Disable appimage launcher by default
env = APPIMAGELAUNCHER_DISABLE,1

# OZONE
env = OZONE_PLATFORM,wayland

# For KVM virtual machines
# env = WLR_NO_HARDWARE_CURSORS, 1
# env = WLR_RENDERER_ALLOW_SOFTWARE, 1

# NVIDIA https://wiki.hyprland.org/Nvidia/
# env = LIBVA_DRIVER_NAME,nvidia
# env = GBM_BACKEND,nvidia-drm
# env = __GLX_VENDOR_LIBRARY_NAME,nvidia
# env = __GL_VRR_ALLOWED,1
# env = WLR_DRM_NO_ATOMIC,1

Config File: /home/******/.config/hypr/conf/input.conf: Read Succeeded
# For all categories, see https://wiki.hyprland.org/Configuring/Variables/
input {
    kb_layout = us
    kb_variant =
    kb_model =
    kb_options =
    kb_rules =

    follow_mouse = 1

    touchpad {
        natural_scroll = true
	clickfinger_behavior = true
	drag_lock = true
    }

    sensitivity = 0 # -0.8 - 0.8, 0 means no modification.
}


Config File: /home/******/.config/hypr/conf/general.conf: Read Succeeded
# See https://wiki.hyprland.org/Configuring/Variables/ for more
general {
    gaps_in = 5
    gaps_out = 20
    border_size = 2
    col.active_border = rgba(33ccffee) rgba(00ff99ee) 45deg
    col.inactive_border = rgba(595959aa)
    layout = dwindle
    resize_on_border = true
}


Config File: /home/******/.config/hypr/conf/decoration.conf: Read Succeeded
# See https://wiki.hyprland.org/Configuring/Variables/ for more
decoration {
    rounding = 10
    blur {
        enabled = true
        size = 3
        passes = 1
    }
    drop_shadow = true
    shadow_range = 4
    shadow_render_power = 3
    col.shadow = rgba(1a1a1aee)
}


Config File: /home/******/.config/hypr/conf/animations.conf: Read Succeeded
# Some default animations, see https://wiki.hyprland.org/Configuring/Animations/ for more
animations {
    enabled = true
    bezier = myBezier, 0.05, 0.9, 0.1, 1.05
    animation = windows, 1, 7, myBezier
    animation = windowsOut, 1, 7, default, popin 80%
    animation = border, 1, 10, default
    animation = borderangle, 1, 8, default
    animation = fade, 1, 7, default
    animation = workspaces, 1, 6, default
    first_launch_animation = true
}


Config File: /home/******/.config/hypr/conf/layouts.conf: Read Succeeded
dwindle {
    # See https://wiki.hyprland.org/Configuring/Dwindle-Layout/ for more
    pseudotile = true # master switch for pseudotiling. Enabling is bound to mainMod + P in the keybinds section below
    preserve_split = true # you probably want this
}

master {
    # See https://wiki.hyprland.org/Configuring/Master-Layout/ for more
    # new_status = master
}


Config File: /home/******/.config/hypr/conf/gestures.conf: Read Succeeded
# See https://wiki.hyprland.org/Configuring/Variables/ for more
gestures {
    workspace_swipe = true
}


Config File: /home/******/.config/hypr/conf/misc.conf: Read Succeeded
# See https://wiki.hyprland.org/Configuring/Variables/ for more
misc {
    disable_hyprland_logo = true
    disable_splash_rendering = true
    disable_autoreload = false
    new_window_takes_over_fullscreen = 2
    initial_workspace_tracking = 1
}
plugin:overview:reverseSwipe = false
plugin:overview:drawActiveWorkspace = true
plugin:overview:onBottom = true
plugin:overview:panelHeight = 175
plugin:overview:panelColor = rgba(0, 0, 0, 0)

Config File: /home/******/.config/hypr/conf/windowrules.conf: Read Succeeded
# See https://wiki.hyprland.org/Configuring/Window-Rules/ for more

Config File: /home/******/.config/hypr/conf/binds.conf: Read Succeeded
# Example binds, see https://wiki.hyprland.org/Configuring/Binds/ for more

# SUPER key
$mainMod = SUPER

# Actions
bind = $mainMod, T, exec, alacritty # Open Alactritty
bind = $mainMod, Q, killactive # Close current window
bind = $mainMod, F, exec, ~/.config/ml4w/settings/filemanager.sh # Opens the filemanager
bind = $mainMod SHIFT, T, togglefloating # Toggle between tiling and floating window
bind = $mainMod SHIFT, F, fullscreen # Open the window in fullscreen
bind = $mainMod, space, exec, rofi -show drun # Open rofi
bind = $mainMod, P, pseudo, # dwindle
bind = $mainMod, J, togglesplit, # dwindle
bind = $mainMod, W, exec, ~/.config/ml4w/settings/browser.sh # Opens the browser
bind = $mainMod SHIFT, W, exec, ~/.config/ml4w/settings/browser_p.sh # Opens browser with -P flag.
bind = $mainMod SHIFT, B, exec, ~/.config/ml4w/scripts/reload-waybar.sh # Reload Waybar
bind = $mainMod, L, exec, hyprlock
bind = $mainMod, V, exec, rofi -modi clipboard:~/.config/custom/cliphist-rofi.sh -show clipboard
bind = $mainMod SHIFT, V, exec, cliphist wipe
bind = $mainMod, code:60, exec, rofi -modi emoji -show emoji
bind = $mainMod, code:135, exec, ~/.config/custom/qs.sh

# SwayOSD & Media Keys
binde =, XF86AudioRaiseVolume, exec, swayosd-client --output-volume +2
binde =, XF86AudioLowerVolume, exec, swayosd-client --output-volume -2
bindl =, XF86AudioMute, exec, swayosd-client --output-volume mute-toggle
bindl =, XF86AudioMicMute, exec, swayosd-client --input-volume mute-toggle
binde =, XF86MonBrightnessUp, exec, swayosd-client --brightness +5
binde =, XF86MonBrightnessDown, exec, swayosd-client --brightness -5
bindl =, XF86AudioPlay, exec, playerctl play-pause

# Lid Switch
bindl = , switch:on:Lid Switch, exec, hyprlock

# Move focus with mainMod + arrow keys
bind = $mainMod, left, movefocus, l # Move focus left
bind = $mainMod, right, movefocus, r # Move focus right
bind = $mainMod, up, movefocus, u # Move focus up
bind = $mainMod, down, movefocus, d # Move focus down

# Switch workspaces with mainMod + [0-9]
bind = $mainMod, 1, workspace, 1 # Switch to workspace 1
bind = $mainMod, 2, workspace, 2 # Switch to workspace 2
bind = $mainMod, 3, workspace, 3 # Switch to workspace 3
bind = $mainMod, 4, workspace, 4 # Switch to workspace 4
bind = $mainMod, 5, workspace, 5 # Switch to workspace 5
bind = $mainMod, 6, workspace, 6 # Switch to workspace 6
bind = $mainMod, 7, workspace, 7 # Switch to workspace 7
bind = $mainMod, 8, workspace, 8 # Switch to workspace 8
bind = $mainMod, 9, workspace, 9 # Switch to workspace 9
bind = $mainMod, 0, workspace, 10 # Switch to workspace 10

# Move active window to a workspace with mainMod + SHIFT + [0-9]
bind = $mainMod SHIFT, 1, movetoworkspace, 1 #  Move window to workspace 1
bind = $mainMod SHIFT, 2, movetoworkspace, 2 #  Move window to workspace 2
bind = $mainMod SHIFT, 3, movetoworkspace, 3 #  Move window to workspace 3
bind = $mainMod SHIFT, 4, movetoworkspace, 4 #  Move window to workspace 4
bind = $mainMod SHIFT, 5, movetoworkspace, 5 #  Move window to workspace 5
bind = $mainMod SHIFT, 6, movetoworkspace, 6 #  Move window to workspace 6
bind = $mainMod SHIFT, 7, movetoworkspace, 7 #  Move window to workspace 7
bind = $mainMod SHIFT, 8, movetoworkspace, 8 #  Move window to workspace 8
bind = $mainMod SHIFT, 9, movetoworkspace, 9 #  Move window to workspace 9
bind = $mainMod SHIFT, 0, movetoworkspace, 10 #  Move window to workspace 10

# Scroll through existing workspaces with mainMod + scroll
bind = $mainMod, mouse_down, workspace, e+1 # Scroll workspaces
bind = $mainMod, mouse_up, workspace, e-1 # Scroll workspaces

# Move/resize windows with mainMod + LMB/RMB and dragging
bindm = $mainMod, mouse:272, movewindow # Move window
bindm = $mainMod, mouse:273, resizewindow # Resize window

======Config-End========
```

</details>

### Description

Hi there! I'm new to Hyprland, fairly famili. I set up this system recently to daily drive on my old Microsoft Surface Laptop Go. I used [ML4W's Hyprland Starter](https://github.com/mylinuxforwork/hyprland-starter) and built my customization and custom bits around it. It's been going great, except for one annoying bug. Whenever I touch the screen, the borders of windows return to the default blue and rounding, Hyprshade resets, (if it's running) and Hyprspace stops working. Interestingly, Hyprspace is still affected even if ```disable-autoreload``` is set to true. I've done some research and couldn't find anybody else with this issue. I have looked at the input logs (I think that's what their called?) for the touchscreen but have no idea what they mean. I currently have a fix for the plugin stopping, (Hyprspace runs on exec instead of exec-once) but there are significant frame rate drops whenever it happens which is very inconvenient. 

### How to reproduce

It's likely some kind of niche bug with how the touchscreen is handled or a misconfiguration on my part which will make this hard to reproduce. All I have to do is touch the touchscreen at any time.

**Comment by vaxerski** [comment]
does this happen on the default config? I doubt this is a problem with hyprland?

**Comment by nickodei** [comment]
hey, I just want to add that I have a **Microsoft Surface 8 Pro** with **Hyprland** and Nixos and the touchscreen works as expected with my config (that's close to default). No such artifacts on my side, if the info somehow helps.

**Comment by Welsey0** [comment]
With some further testing (removing my configuration) the issue is resolved, but I am extremely confused on what configuration option might make this happen? Apologies if this was the wrong place to report this kind of issue.

**Comment by Welsey0** [comment]
I may have resolved this wrongly? Can somebody with more knowledge advise me on what to do with this issue? I'm going to mark it as cannot reproduce.

**Comment by Welsey0** [comment]
Apologies if this was the incorrect way to go about solving the issue. I'll provide some info on the outcome:

The default border color and rounding was due to the way ML4W's Hyprland Settings app was built, but had nothing to do with the reloading. It was just an indicator of the reload. Hard-coding settings I had adjusted there fixed that, but not the reloading. I could still tell that I was reloading as my Waybar workspace indicator would flash.

After doing a full system upgrade today the issue is resolved? I did not change my configuration so it must have been an issue in something that got patched.

Anyway, issue is resolved!
